packages/insomnia-app/app/ui/components/modals/settings-modal.js

```diff
@@ -65,11 +65,36 @@ class SettingsModal extends PureComponent {
     this.modal.hide();
   }
 
-  async _handleChangeTheme(theme, persist = true) {
-    await setTheme(theme);
+  async _handleChangeTheme(themeName, colorScheme, persist = true) {
+    const { settings } = this.props;
+
+    let patch;
+    switch (colorScheme) {
+      case 'light':
+        patch = { theme: themeName };
+        break;
+      case 'dark':
+        patch = { darkTheme: themeName };
+        break;
+      default:
+        patch = { theme: themeName };
+        break;
```

Comment:
It might be good to add the case for `'default'` as well, because `ColorScheme` is typed as `'default' | 'light' | 'dark'`

```suggestion
      case 'default':
      default:
        patch = { theme: themeName };
        break;
```